crypto/conf/conf.c

```diff
@@ -642,6 +642,18 @@ int CONF_modules_load_file(const char *filename, const char *appname,
   return 1;
 }
 
+char *CONF_get1_default_config_file(void) {
+  const char *temp = "No support for Config files in AWS-LC.";
+  size_t temp_len = strlen(temp) + 1;
+
+  char *ret = (char *)OPENSSL_malloc(temp_len);
```

Comment:
When you create a compile time constant string like this is it guaranteed that temp is null terminated so strlen will work as expected? Also when you malloc temp_len there might not be a 0 at the end so after the memcpy there could be garbage at the end too?

Reply:
For the first question, yes.
* https://en.cppreference.com/w/c/language/string_literal
* https://stackoverflow.com/questions/47288881/are-c-constant-character-strings-always-null-terminated

For the second question, I'm not entirely sure what you mean by 0 coming into play. The additional `+ 1` is to account for the null terminator. This follows a pattern [in OpenSSL](https://github.com/openssl/openssl/blob/0285160ffa3b8c2b5491222243042593808298c4/crypto/conf/conf_mod.c#L697) and [our codebase](https://github.com/aws/aws-lc/blob/94e91d96d311404a0eaaca6039bfd7e6577c6dff/crypto/mem.c#L382) as well.

Good news I discovered `OPENSSL_strdup` while fiddling with this. I'll call this instead of all the duplicate hocus pocus here.